### PR TITLE
Mc inscale porting

### PR DESCRIPTION
**Pull request by lwyszomi**
@fyniac

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 5 errors:

* custom/reports/mc/models.py, line 123 - E501 line too long (118 > 115 characters)
* custom/reports/mc/reports/sql.py, line 70 - E301 expected 1 blank line, found 0
* custom/reports/mc/reports/sql.py, line 423 - E241 multiple spaces after ','
* custom/reports/mc/reports/sql.py, line 553 - E241 multiple spaces after ','
* custom/reports/mc/reports/sql.py, line 676 - W292 no newline at end of file

File: custom/reports/mc/reports/sql.py [hunk 1]
```diff
@@ -1,510 +1,685 @@
...
+                AliasColumn('diagnosed_diarrhea'),
+                AliasColumn('diagnosed_ari')
+            ], slug='diagnosed_total'),
+            DatabaseColumn(_('treated_malaria'),
+                           CountColumn('doc_id', alias='treated_malaria', filters=self.filters + [OR([
+                               AND([EQ('it_malaria_child', 'one'), EQ('malaria_child', 'one')]),
+                               AND([EQ('it_malaria_adult', 'one'), EQ('malaria_adult', 'one')])
+                           ])])),
+            DatabaseColumn(_('treated_diarrhea'),
+                           CountColumn('doc_id',  alias='treated_diarrhea', filters=self.filters + [OR([
```
Comment (dimagimon) [inline custom/reports/mc/reports/sql.py]: E241 multiple spaces after ','

File: custom/reports/mc/reports/sql.py [hunk 2]
```diff
@@ -1,510 +1,685 @@
...
+        return [
+            DatabaseColumn('Date', SimpleColumn('date')),
+            DatabaseColumn('hf', SimpleColumn('hf')),
+            DatabaseColumn('district', SimpleColumn('district')),
+            DatabaseColumn(_('form/stock/stock_amox_pink'), SimpleColumn('stock_amox_pink')),
+            DatabaseColumn(_('form/stock/stock_amox_green'), SimpleColumn('stock_amox_green')),
+            DatabaseColumn(_('form/stock/stock_ors'), SimpleColumn('stock_ors')),
+            DatabaseColumn(_('form/stock/stock_ra_50'), SimpleColumn('stock_ra_50')),
+            DatabaseColumn(_('form/stock/stock_ra_200'), SimpleColumn('stock_ra_200')),
+            DatabaseColumn(_('form/stock/stock_zinc'),  SimpleColumn('stock_zinc')),
```
Comment (dimagimon) [inline custom/reports/mc/reports/sql.py]: E241 multiple spaces after ','

**Comment by czue** [comment]
@lwyszomi - haven't reviewed too closely yet but one thought - would it be possible to separate the additional report from the deletion of the existing code? that will make it super easy to do a side-by-side comparison on prod and confirm that everything is working before deleting the old code.

we can put the new report behind the "visible to only UCR" feature flag so that we can iterate on that while not affecting the existing project

**Comment by lwyszomi** [comment]
@czue This is possible, I will update PR today.

**Comment by lwyszomi** [comment]
@cory Where can I find "visible to only UCR" feature flag?

**Comment by czue** [comment]
sorry, it's just `USER_CONFIGURABLE_REPORTS` + `visible=False`

**Comment by dimagimon** [comment]
There are 20 errors:

* custom/reports/mc/models.py, line 627 - W292 no newline at end of file
* custom/reports/mc/reports/old_definitions.py, line 198 - E501 line too long (175 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 199 - E501 line too long (210 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 200 - E501 line too long (208 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 201 - E501 line too long (215 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 202 - E501 line too long (210 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 203 - E501 line too long (195 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 204 - E501 line too long (192 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 205 - W292 no newline at end of file
* custom/reports/mc/reports/old_sql.py, line 95 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 212 - E303 too many blank lines (2)
* custom/reports/mc/reports/old_sql.py, line 281 - E501 line too long (148 > 115 characters)
* custom/reports/mc/reports/old_sql.py, line 293 - E303 too many blank lines (2)
* custom/reports/mc/reports/old_sql.py, line 347 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 353 - E303 too many blank lines (2)
* custom/reports/mc/reports/old_sql.py, line 425 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 435 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 445 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 489 - E301 expected 1 blank line, found 0
* custom/reports/mc/reports/old_sql.py, line 510 - W292 no newline at end of file

**Comment by lwyszomi** [comment]
@czue updated, please don't look on dimagimoon because this is old code.

**Comment by dimagimon** [comment]
There are 20 errors:

* custom/reports/mc/models.py, line 627 - W292 no newline at end of file
* custom/reports/mc/reports/old_definitions.py, line 198 - E501 line too long (175 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 199 - E501 line too long (210 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 200 - E501 line too long (208 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 201 - E501 line too long (215 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 202 - E501 line too long (210 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 203 - E501 line too long (195 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 204 - E501 line too long (192 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 205 - W292 no newline at end of file
* custom/reports/mc/reports/old_sql.py, line 95 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 212 - E303 too many blank lines (2)
* custom/reports/mc/reports/old_sql.py, line 281 - E501 line too long (148 > 115 characters)
* custom/reports/mc/reports/old_sql.py, line 293 - E303 too many blank lines (2)
* custom/reports/mc/reports/old_sql.py, line 347 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 353 - E303 too many blank lines (2)
* custom/reports/mc/reports/old_sql.py, line 425 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 435 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 445 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 489 - E301 expected 1 blank line, found 0
* custom/reports/mc/reports/old_sql.py, line 510 - W292 no newline at end of file

**Comment by czue** [comment]
@lwyszomi this is a pretty hard diff to read, but I assume the `old_` files are just copy/paste of the previous implementation and that you're quite confident it didn't break? If so then I think we can merge and then QA the new reports on prod

File: corehq/apps/userreports/operators.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 NOT_EQUAL = lambda input, reference: input != reference
 IN = lambda input, reference_list: input in reference_list
 IN_MULTISELECT = lambda input, reference: reference in (input or '').split(' ')
+ANY_IN_MULTISELECT = lambda input, reference: any([subval in (input or '').split(' ') for subval in reference])
```
Comment (czue) [inline corehq/apps/userreports/operators.py]: did you end up using this? didn't see any other references

Comment (lwyszomi) [inline corehq/apps/userreports/operators.py]: @czue I added this in operators dict https://github.com/dimagi/commcare-hq/pull/8842/files#diff-0da33b434582bbcf21b2db76d7da5b6aR20

**Comment by lwyszomi** [comment]
@czue I know that is hard diff, you have right the ```_old``` files are copy/paste. I tested old reports and all works fine.

File: corehq/apps/userreports/operators.py (hunk 3 above)
Comment (czue) [inline corehq/apps/userreports/operators.py]: but do the reports use this operator?

**Comment by czue** [comment]
just waiting for you to confirm how the `in_multi` is being used and then good to merge

File: corehq/apps/userreports/operators.py (hunk 3 above)
Comment (lwyszomi) [inline corehq/apps/userreports/operators.py]: @czue 
Yes reports use this.

Operator which I added works: For example in xforms we have property ```test = "1 2 3 4 5"``` and in reports we have calculator with condition (if any value from ```test``` in ```[2, 3]```), then we check if any of values from ```test``` property are in the list ```[2, 3]```, but ```in_mulit``` we only can check one value.

Is that make sanse?

**Comment by czue** [comment]
looks like you could simplify the data source a lot with named filters and expressions.

can you change the `engine_id` of both data sources to `"ucr"`? then I think good to go.

**Comment by dimagimon** [comment]
There are 20 errors:

* custom/reports/mc/models.py, line 627 - W292 no newline at end of file
* custom/reports/mc/reports/old_definitions.py, line 198 - E501 line too long (175 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 199 - E501 line too long (210 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 200 - E501 line too long (208 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 201 - E501 line too long (215 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 202 - E501 line too long (210 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 203 - E501 line too long (195 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 204 - E501 line too long (192 > 115 characters)
* custom/reports/mc/reports/old_definitions.py, line 205 - W292 no newline at end of file
* custom/reports/mc/reports/old_sql.py, line 95 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 212 - E303 too many blank lines (2)
* custom/reports/mc/reports/old_sql.py, line 281 - E501 line too long (148 > 115 characters)
* custom/reports/mc/reports/old_sql.py, line 293 - E303 too many blank lines (2)
* custom/reports/mc/reports/old_sql.py, line 347 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 353 - E303 too many blank lines (2)
* custom/reports/mc/reports/old_sql.py, line 425 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 435 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 445 - E302 expected 2 blank lines, found 1
* custom/reports/mc/reports/old_sql.py, line 489 - E301 expected 1 blank line, found 0
* custom/reports/mc/reports/old_sql.py, line 510 - W292 no newline at end of file

**Comment by lwyszomi** [comment]
@czue 
I changed the ```engine_id```.

**Comment by czue** [comment]
we'll see how UCR does with 2000 lines of JSON!

**Comment by snopoke** [comment]
woah

On 30 October 2015 at 16:20, Cory Zue <notifications@github.com> wrote:

> we'll see how UCR does with 2000 lines of JSON!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/8842#issuecomment-152538805>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Comment by snopoke** [comment]
would named expressions make it smaller?

**Comment by czue** [comment]
yes, I think by quite a bit. named filters too. @lwyszomi do you think you can take a crack at that?

**Comment by lwyszomi** [comment]
@czue  I can do this.

**Pull request merged**